backend/src/scripts/utility/teamspaces/removeIncompleteRevisions.js

```diff
@@ -0,0 +1,214 @@
+			`${supermeshId}_normals`,
+			`${supermeshId}_vertices`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.json_mpc.ref`, { _id: { $in: [
+			`${supermeshId}.json.mpc`,
+			`${supermeshId}_unity.json.mpc`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.src.ref`, { _id: { $in: [
+			`${supermeshId}.src.mpc`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.unity3d.ref`, { _id: { $in: [
+			`${supermeshId}.unity3d`,
+		] } }));
+
+		return supermeshId;
+	});
+
+	modelStashPromises.push(removeRecords(teamspace, `${model}.stash.unity3d`, { _id: revId }));
+	modelStashPromises.push(removeRecords(teamspace, `${model}.stash.3drepo`, { rev_id: revId }, '_extRef'));
+
+	await Promise.all(modelStashPromises);
+};
+
+const processModelSequences = async (teamspace, model, revId) => {
+	const sequences = await find(teamspace, `${model}.sequences`, { rev_id: revId }, { frames: 1 });
+	const sequenceFilenames = [];
+
+	for (const { _id, frames } of sequences) {
+		if (frames) {
+			for (const { state } of frames) {
+				sequenceFilenames.push(state);
+			}
+		}
+
+		// eslint-disable-next-line no-await-in-loop
+		await Promise.all([
+			removeRecords(teamspace, `${model}.activities`, { sequenceId: _id }, '_extRef'),
+			processFilesAndRefs(teamspace, `${model}.activities.ref`, { _id: UUIDToString(_id) }),
+		]);
+	}
+
+	await processFilesAndRefs(teamspace, `${model}.sequences.ref`, { _id: { $in: sequenceFilenames } });
+
+	await deleteMany(teamspace, `${model}.sequences`, { rev_id: revId });
+};
+
+const processModelScene = async (teamspace, model, revId) => {
+	await removeRecords(teamspace, `${model}.scene`, { rev_id: revId }, '_extRef');
+};
+
+const removeRevision = async (teamspace, model, revId) => {
+	logger.logInfo(`\t\t-${model}::${UUIDToString(revId)}`);
+
+	await Promise.all([
+		processModelScene(teamspace, model, revId),
+		processModelSequences(teamspace, model, revId),
+		processModelStash(teamspace, model, revId),
+	]);
+};
+
+const processTeamspace = async (teamspace, revisionAge) => {
+	const cols = await getCollectionsEndsWith(teamspace, '.history');
+
+	for (const { name } of cols) {
+		const incompleteRevisionFilter = {
+			incomplete: { $exists: true },
+			timestamp: { $lt: new Date(new Date().setDate(new Date().getDate() - revisionAge)) },
+		};
+		// eslint-disable-next-line no-await-in-loop
+		const badRevisions = await find(teamspace, name, incompleteRevisionFilter, { rFile: 1 });
+
+		for (const { _id, rFile } of badRevisions) {
+			const model = name.slice(0, -('.history'.length));
+			// eslint-disable-next-line no-await-in-loop
+			await removeRevision(teamspace, model, _id);
+
+			// eslint-disable-next-line no-await-in-loop
+			await processFilesAndRefs(teamspace, name, { _id: { $in: rFile } });
+			// eslint-disable-next-line no-await-in-loop
+			await removeRecords(teamspace, name, incompleteRevisionFilter);
```

Comment:
I think this is in the wrong place?

As far as I can tell we're removing all the revisions we've found everytime we loop through one revision.